cmake/FetchHolohubOperator.cmake

```diff
@@ -57,12 +63,25 @@ function(fetch_holohub_operator OPERATOR_NAME)
       DOWNLOAD_COMMAND
         git clone --depth 1 --no-checkout ${ARGS_REPO_URL} "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src"
         && cd "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src"
-        && git sparse-checkout init --cone
-        && git sparse-checkout set operators/${ARGS_PATH}
+        && git sparse-checkout set --no-cone
+            operators/${ARGS_PATH}
+            cmake/pybind11_add_holohub_module.cmake
+            cmake/pybind11/
+            cmake/pydoc/
+            operators/operator_util.hpp
         && git checkout ${ARGS_BRANCH}
-        && mv "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/operators/${ARGS_PATH}" "${FETCHCONTENT_BASE_DIR}/tmp_${OPERATOR_NAME}"
-        && rm -rf "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src"
-        && mv "${FETCHCONTENT_BASE_DIR}/tmp_${OPERATOR_NAME}" "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/"
+        # Write a CMakeLists.txt in the operators directory to set CMAKE_MODULE_PATH and add the operator subdirectory
+        && echo "list(APPEND CMAKE_MODULE_PATH \\$\\{CMAKE_CURRENT_LIST_DIR\\}/cmake)" > "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        # Add code to create python module output directory if building python
+        && echo "if(HOLOHUB_BUILD_PYTHON)" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "  if(NOT CMAKE_INSTALL_LIBDIR)" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "    set(CMAKE_INSTALL_LIBDIR lib)" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "  endif()" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "  set(HOLOHUB_PYTHON_MODULE_OUT_DIR \\$\\{CMAKE_BINARY_DIR\\}/python/\\$\\{CMAKE_INSTALL_LIBDIR\\}/holohub)" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "  file(MAKE_DIRECTORY \\$\\{HOLOHUB_PYTHON_MODULE_OUT_DIR\\})" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "endif()" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
+        && echo "add_subdirectory(operators/${ARGS_PATH})" >> "${FETCHCONTENT_BASE_DIR}/holohub_${OPERATOR_NAME}-prefix/src/CMakeLists.txt"
```

Comment:
Can we compose this into a single "echo" command rather than splitting over multiple calls, in case there is an interruption during setup?